### PR TITLE
Refactor register API

### PR DESCRIPTION
**Pull request by sharath2727**
Description: Refactoring the existing implementation to be able to do specialized operation depending on the Application type. This makes it easy to even introduce overloaded Register API that takes in assets to directly call into appropriate helpers to assist in registration process.

Validation: Run TAEF tests

### CODE REVIEW

File: dev/AppNotifications/AppNotificationManager.cpp [hunk 1]
```diff
@@ -126,50 +126,21 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
...
-                }
+                registeredClsid = RegisterPackagedAppHelper();
             }
-
-            winrt::guid registeredClsid{ GUID_NULL };
-            if (AppModel::Identity::IsPackagedProcess())
+            else
             {
-                registeredClsid = PushNotificationHelpers::GetComRegistrationFromRegistry(expectedAppServerArgs.data());
+                AppNotificationAssets assets{ GetAssetsHelper() };
```
Comment (danielayala94) [inline dev/AppNotifications/AppNotificationManager.cpp:136]: No need of naming as Helper I think.

File: dev/AppNotifications/AppNotificationManager.cpp [hunk 2]
```diff
@@ -178,6 +149,63 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
...
+        }
+
+        return registeredClsid;
+    }
+
+    winrt::guid AppNotificationManager::RegisterUnpackagedAppHelper(std::wstring const& displayName, std::wstring const& iconFilePath)
+    {
+        THROW_IF_FAILED(PushNotifications_RegisterFullTrustApplication(m_appId.c_str(), GUID_NULL));
+
+        // GuidString is registered in
```
Comment (danielayala94) [inline dev/AppNotifications/AppNotificationManager.cpp:194]: registered where?

File: dev/AppNotifications/AppNotificationUtility.cpp [hunk 3]
```diff
@@ -264,14 +266,27 @@ winrt::guid Microsoft::Windows::AppNotifications::Helpers::RegisterComActivatorG
...
-    return winrt::guid(registeredGuid.substr(1, registeredGuid.size() - 2));
+// Try the following techniques to retrieve display name and icon:
+// 1. Based on the best app shortcut, using the FrameworkUdk.
+// 2. From the current process.
+// 3. Set a default DisplayName, but leave empty the icon file path so Shell can set a default icon.
+AppNotificationAssets Microsoft::Windows::AppNotifications::Helpers::GetAssetsHelper()
+{
+    AppNotificationAssets assets{};
+
+    if (FAILED(Microsoft::Windows::AppNotifications::ShellLocalization::RetrieveAssetsFromShortcut(assets)) &&
```
Comment (danielayala94) [inline dev/AppNotifications/AppNotificationUtility.cpp:280]: Can you please use a namespace for ShellLocalization? These lines are unnecessarily long.

**Comment by loneursid** [comment]
Why wasn't this PR open against "main" and not "develop" as per the new guidance?

**Comment by sharath2727** [comment]
Closing the PR and will open the PR against develop branch as per the guidelines.

**Pull request closed without merging**